### PR TITLE
feat: Custom https cert

### LINKED ISSUE #2118: 🐛 BUG: Local dev mode with https generates a new self signed cert every time
### What version of `Wrangler` are you using?

2.1.14

### What operating system are you using?

Mac

### Describe the Bug

When using wrangler to test workers, I need to use https, and use --local-protocol https, well actually I specify it in the toml with:
```
[dev]
local_protocol = "https"
```
The problem is that when running in local mode, miniflare is used and it generates a self signed cert every single time I launch. This message is output each time:
```
[mf:inf] Generating new self-signed certificate...
```

A freshly generated self-signed cert every time is awkward because most modern browsers will reject them.

When not in local mode, the wrangler code creates/reads the cert, and will default to ~/.wrangler/local-cert
For my use case I replace the cert in this folder, but I can't do this miniflare when it's running in a temp dir.

Also, I would argue that the two modes should use the same cert, and from a look at the code it looks like miniflare can be passed the path to the cert and key (options https-key and https-cert).

My personal preference would be to be able to optionally be able to pass the cert & key files through to wrangler, and it also forward those on to miniflare in local mode.

### PR DESCRIPTION
**Pull request by paulrostorp**
Fixes #2118

**What this PR solves / how to test:**
Add flags `--https-key-path` and `--https-cert-path` to wrangler pages dev command

**Author has addressed the following:**

- Tests
  - [x] Included (although the actual feature has only been tested manually by @petebacondarwin)
  - [ ] Not necessary because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [x] Included
  - [ ] Not necessary because:
- Associated docs
  - [ ] Issue(s)/PR(s): https://github.com/cloudflare/cloudflare-docs/issues/13032
  - [ ] Not necessary because:

### CODE REVIEW

**Comment by lrapoport-cf** [comment]
blocked for now (see https://github.com/cloudflare/workers-sdk/issues/2118#issuecomment-1834602329)

**Comment by petebacondarwin** [comment]
I know this is blocked for the moment, but I would like to propose a simpler implementation, for when we are in a position to land this. Instead of using command line arguments, which need to be threaded down through the various calls. How about we just have a pair of environment variables, e.g. `CLOUDFLARE_LOCAL_HTTPS_KEY_PATH` and `CLOUDFLARE_LOCAL_HTTPS_CERT_PATH`.
Then the implementation would simply check these environment variables in the `getHttpsOptions()` function before attempting to generate these files.

Is there a reason (e.g. prior art) for doing this via command line arguments that I am missing?

**Comment by petebacondarwin** [comment]
> @petebacondarwin I did consider that as it was a much simpler implementation, but considering this CLI aims to support configuration both through flags and a config file (wrangler.toml), I thought it best not to break that pattern.

Ah yes and looking through the original issue it seems that there is value in being able to set different certs for each wrangler instance, which is more awkward using env vars - https://github.com/cloudflare/workers-sdk/issues/2118#issuecomment-1382890616

**Comment by petebacondarwin** [comment]
I rebased and fixed up this branch. Will do a proper review later this week.

File: packages/wrangler/src/api/startDevWorker/types.ts [hunk 1]
```diff
@@ -81,7 +81,13 @@ export interface StartDevWorkerOptions {
...
 
 		/** The local address to reach your worker. Applies to remote: true (remote mode) and remote: false (local mode). */
-		server?: { hostname?: string; port?: number; secure?: boolean }; // hostname: --ip, port: --port, secure: --local-protocol
+		server?: {
+			hostname?: string;
+			port?: number;
+			secure?: boolean;
+			httpsKeyPath?: string;
+			httpsCertPath?: string;
+		}; // hostname: --ip, port: --port, secure: --local-protocol
```
Comment (petebacondarwin) [inline packages/wrangler/src/api/startDevWorker/types.ts:90]:
```suggestion
			hostname?: string; // --ip
			port?: number; // --port
			secure?: boolean; // --local-protocol
			httpsKeyPath?: string;
			httpsCertPath?: string;
		};
```

**Review by petebacondarwin — APPROVED** [review]

**Review by penalosa — APPROVED** [review]

**Comment by Aphanite** [comment]
Thanks a lot for this and for including it in the latest release! However, when I searched for the first version to support it, I found no mention of it in the recent releases.

**Comment by petebacondarwin** [comment]
Gosh! You are right. That is my mistake for landing this without a changeset. It was first available in 3.28.3. I'll update the changelog accordingly.

**Comment by petebacondarwin** [comment]
https://github.com/cloudflare/workers-sdk/pull/5147

**Pull request merged**